src/options.cc

```diff
@@ -16,12 +24,12 @@ bool Options::loadOptionsFromStream(std::istream* config_file) {
 	for (unsigned int i =0; i < config.statements_.size();i++) {
 
 		// Sets the port
-		if (!issetPort && config.statements_[i]->tokens_[0] == "server") {
+		if (!issetPort && config.statements_[i]->tokens_[COM] == "server") {
 			std::shared_ptr<NginxConfigStatement> temp_config = config.statements_[i];
 			for (unsigned int j = 0; j < temp_config->child_block_->statements_.size(); j++) {
-				if(temp_config->child_block_->statements_[j]->tokens_[0] == "listen") {
-					std::string port = temp_config->child_block_->statements_[j]->tokens_[1];
-					if((unsigned int) std::stoi(port) > 65535 || (unsigned int) std::stoi(port) < 1024) {
+				if(temp_config->child_block_->statements_[j]->tokens_[COM] == "listen") {
```

Comment:
We might need to check `tokens_.size()`, I think a config like
```
server {
    listen;
}
```
might crash? could be a good unit test to try out.